agent/data/client.go

```diff
@@ -43,6 +44,7 @@ var (
 	buckets = []string{
 		imagesBucketName,
 		containersBucketName,
+		pulledContainersBucketName,
```

Comment:
is a separate bucket for pulled container necessary? each container is already added to the `containers` bucket when it transitions to pulled: https://github.com/aws/amazon-ecs-agent/blob/478c4ec00a43b4713a0acc811cfd1e2d70307f63/agent/engine/task_manager.go#L422. saving it to another bucket means adding an extra save for every container while it doesn't save anything new. To ensure that pulled containers can be correctly loaded into state when agent restarts, it seems to me that you don't need anything change for container saving/removing, while for loading, you can make a change here https://github.com/aws/amazon-ecs-agent/blob/478c4ec00a43b4713a0acc811cfd1e2d70307f63/agent/engine/data.go#L68-L77 to check the container status, and add pulled container to state with `state.AddPulledContainer` while adding other containers with `state.AddContainer`